File: sensor_controller_py/sensor_controller_py/data_publisher.py
import rclpy
from rclpy.node import Node

from std_msgs.msg import Int64


class DataPublisher(Node):

    def __init__(self):
        super().__init__('data_publisher')
        self.publisher_ = self.create_publisher(Int64, 'sensor_data', 10)

        self.timer = self.create_timer(2, self.timer_callback)
        self.i = 0

    def timer_callback(self):
        msg = Int64()
        msg.data = self.i
        self.publisher_.publish(msg)
        
        self.get_logger().info(f"Publishing: '{msg.data}'")
        self.i += 1


def main(args=None):
    rclpy.init(args=args)

    data_publisher = DataPublisher()

    rclpy.spin(data_publisher)

    data_publisher.destroy_node()
    rclpy.shutdown()


if __name__ == '__main__':
    main()

File: sensor_controller_cpp/src/data_publisher.cpp
#include <chrono>

#include "rclcpp/rclcpp.hpp"
#include "std_msgs/msg/int64.hpp"

using namespace std::chrono_literals;

class DataPublisher : public rclcpp::Node
{
  public:
    DataPublisher()
    : Node("data_publisher")
    {
      publisher_ = this->create_publisher<std_msgs::msg::Int64>("sensor_data", 10);
      timer_ = this->create_wall_timer(2s, std::bind(&DataPublisher::timer_callback, this));
    }

  private:
    void timer_callback()
    {
      auto message = std_msgs::msg::Int64();
      message.data = count_++;
      
      RCLCPP_INFO(this->get_logger(), "Publishing: '%ld'", message.data);
      publisher_->publish(message);
    }

    rclcpp::TimerBase::SharedPtr timer_;
    rclcpp::Publisher<std_msgs::msg::Int64>::SharedPtr publisher_;
    size_t count_;
};

int main(int argc, char * argv[])
{
  rclcpp::init(argc, argv);

  rclcpp::spin(std::make_shared<DataPublisher>());
  
  rclcpp::shutdown();
  return 0;
}